### PR TITLE
Added initial implementation of DynamicsCRMHelper

### PR DESCRIPTION
**Pull request by DigitalFlow**
Hey all,

I'm currently implementing a new helper for Dynamics CRM.
The use of it is to be able to export solutions (update component containers) from Dynamics CRM and being able to store those solutions in a VCS or even import them before doing unit tests for ensuring that the system is in the correct state for doing the unit tests.
Tests as well as a possibility for downlevelling the solution archive into xml representation will follow.

Kind Regards,
DigitalFlow

### CODE REVIEW

**Comment by DigitalFlow** [comment]
What I was wondering about:
Is it ok that I added the NuGet Packages for CRM SDK? I use them for connecting to the CRM services.

**Comment by forki** [comment]
Aehm that's not a good idea

**Comment by DigitalFlow** [comment]
Ok, so I'll remove those packages and connect differently. I wonder why the mono build on Travis is failing, while the AppVeyor build with MSBuild works...

**Comment by DigitalFlow** [comment]
@forki: How sounds using Fsharp Data TypeProviders for connecting to the CRM to you? Seems if I'd need them as NuGet packages too however.
https://www.nuget.org/packages/FSharp.Data/2.2.5/
https://www.nuget.org/packages/FSharp.Data.TypeProviders/

**Comment by forki** [comment]
we do we need to add packages at all?

**Comment by DigitalFlow** [comment]
Ok so I checked my options and I don't see an alternative to the CRM packages that does not produce a lot of overhead for me. 
Shall we close the Pull Request in this case?

**Comment by forki** [comment]
mhm. I think we should create an addition project then. We already have way to much stuff in this repo.

**Comment by DigitalFlow** [comment]
All right then

**Comment by forki** [comment]
sorry. Is there a command line version we can call from here?

**Comment by DigitalFlow** [comment]
I just began developing a command line version, I'll check with my boss whether I am allowed to open source it. Are you thinking of making it a complete standalone tool, or a subproject of FAKE?

**Comment by forki** [comment]
What we usually do is to create only a wrapper around the command line in fake (That stuff that calls the command line tool),

**Comment by DigitalFlow** [comment]
Yes thats what I did for the SolutionPackager part, but I guess you mean for this whole module then.
I'll have to develop a command line version anyhow, I would just have to check with my boss for open sourcing it and writing a FAKE wrapper.

**Comment by DigitalFlow** [comment]
Ok so it goes on. I am allowed to open source the tool.
The plan is to leave solution packager part of this pull request as it is, because it does not need any packages. For the importing and exporting I will write a small command line tool and a wrapper for FAKE so that we can use the wrapper and don't need any nuget packages in here.

**Comment by DigitalFlow** [comment]
Ok, my command line tool is done for now: https://github.com/DigitalFlow/Dynamics-CRM-Solution-Exchanger
The wrapper is already working in this PR, I already set up a build in our CI which exports a solution from CRM every night and pushes it into our VCS.

**Comment by DigitalFlow** [comment]
@forki: Can you imagine why the build is failing? Local builds run just fine, I didn't add any assemblies that would need to be referenced and the previous build passed at least for the previous commit, while I only changed an XML tag in the current commit...

**Comment by forki** [comment]
seems unrelated. mono fails with mono 4.2 issue and appveyor with nuget.

File: src/app/FakeLib/DynamicsCRMHelper.fs [hunk 1]
```diff
@@ -0,0 +1,143 @@
+﻿[<AutoOpen>]
```
Comment (forki) [inline src/app/FakeLib/DynamicsCRMHelper.fs]: please remove the autoopen

**Comment by DigitalFlow** [comment]
Ok, I removed the auto open tag.

**Comment by DigitalFlow** [comment]
Ok, so I just added some documentation and some comments.
Do you need anything else for it being merged?

**Comment by forki** [comment]
cool. thx

**Pull request merged**